### PR TITLE
List all amp javascript files needed

### PR DESCRIPTION
**Pull request by towanda**

### CODE REVIEW

File: spec/article_json/export/amp/exporter_spec.rb [hunk 1]
```diff
@@ -9,4 +9,29 @@
...
+        'src="https://cdn.ampproject.org/v0/amp-vimeo-0.1.js"></script>',
+        '<script async custom-element="amp-youtube" ' \
+        'src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>',
+        '<script async custom-element="amp-facebook" ' \
+        'src="https://cdn.ampproject.org/v0/amp-facebook-0.1.js"></script>',
+        '<script async custom-element="amp-iframe"' \
+        'src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>',
+        '<script async custom-element="amp-twitter" ' \
+        'src="https://cdn.ampproject.org/v0/amp-twitter-0.1.js"></script>'
+      ]
```
Comment (christofdamian) [inline spec/article_json/export/amp/exporter_spec.rb:30]: is the order in a Set defined?

File: lib/article_json/export/amp/exporter.rb [hunk 2]
```diff
@@ -16,6 +17,16 @@ def html
           end
           doc.to_html(save_with: 0)
         end
+
+        def amp_libraries
```
Comment (dsager) [inline lib/article_json/export/amp/exporter.rb]: can you please add some method documentation?! thanks!

Comment (towanda) [reply lib/article_json/export/amp/exporter.rb]: sure!

File: lib/article_json/export/amp/elements/embed.rb [hunk 3]
```diff
@@ -4,22 +4,16 @@ module AMP
       module Elements
         class Embed < Base
           include Shared::Caption
+          attr_reader :amp_library
```
Comment (dsager) [inline lib/article_json/export/amp/elements/embed.rb]: what do you think about implementing this as a method:
```ruby
def amp_library
  return @amp_library if defined? @amp_library
  type_specific_node # the execution will determine needed amp library
  @amp_library
end
```

like this we would keep `#type_specific_node` private and the logic of when/how stuff is calculated withing this method...

File: spec/article_json/export/amp/exporter_spec.rb (hunk 1 above)
Comment (nicolas-fricke) [reply spec/article_json/export/amp/exporter_spec.rb:30]: Nope, the order is uncertain:

> Set implements a collection of unordered values with no duplicates.

https://ruby-doc.org/stdlib-2.4.2/libdoc/set/rdoc/Set.html (very first line :wink: )

File: lib/article_json/export/amp/elements/embed.rb (hunk 3 above)
Comment (towanda) [reply lib/article_json/export/amp/elements/embed.rb]: I like it, changing it

File: spec/article_json/export/amp/exporter_spec.rb (hunk 1 above)
Comment (christofdamian) [reply spec/article_json/export/amp/exporter_spec.rb:30]: then one should probably not compare these two arrays with ```eq``` below.

Comment (dsager) [reply spec/article_json/export/amp/exporter_spec.rb:30]: right, `match_array` is what we should use.

Comment (towanda) [reply spec/article_json/export/amp/exporter_spec.rb:30]: changed, thanks!

**Comment by towanda** [comment]
@dsager I added the changes you suggested.

**Comment by christofdamian** [comment]
+1

**Comment by dsager** [comment]
+1
nice, thanks!

**Pull request merged**